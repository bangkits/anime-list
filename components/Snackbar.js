import { Snackbar, Alert } from "@mui/material";

const SnackbarComponent = (props) => {
  const { type, message, show, onDismiss } = props;

  return (
    <Snackbar
      anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
      autoHideDuration={3000}
      open={show}
      onClose={onDismiss}
    >
      <Alert severity={type}>{message}</Alert>
    </Snackbar >
  )
};

SnackbarComponent.defaultProps = {
  type: 'success',
  message: '',
  show: false,
  onDismiss: () => { }
};

export default SnackbarComponent;