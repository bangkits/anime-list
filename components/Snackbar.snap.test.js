import renderer from 'react-test-renderer';
import Snackbar from './Snackbar';

describe('Snackbar Component Snapshot', () => {
  it('Renders success snackbar component', () => {
    const props = {
      type: 'success',
      message: 'message',
      show: true,
      onDismiss: jest.fn()
    };

    // prevent Cannot read property 'scrollTop' of null scrollTop of null error
    const tree = renderer.act(() => {
      renderer.create(<Snackbar {...props} />, {
        createNodeMock: node => {
          return document.createElement(node.type);
        },
      });
    });
    expect(tree).toMatchSnapshot();
  });

  it('Renders error snackbar component', () => {
    const props = {
      type: 'error',
      message: 'message',
      show: true,
      onDismiss: jest.fn()
    };

    // prevent Cannot read property 'scrollTop' of null scrollTop of null error
    const tree = renderer.act(() => {
      renderer.create(<Snackbar {...props} />, {
        createNodeMock: node => {
          return document.createElement(node.type);
        },
      });
    });
    expect(tree).toMatchSnapshot();
  });

  it('Renders warning snackbar component', () => {
    const props = {
      type: 'warning',
      message: 'message',
      show: true,
      onDismiss: jest.fn()
    };

    // prevent Cannot read property 'scrollTop' of null scrollTop of null error
    const tree = renderer.act(() => {
      renderer.create(<Snackbar {...props} />, {
        createNodeMock: node => {
          return document.createElement(node.type);
        },
      });
    });
    expect(tree).toMatchSnapshot();
  });
});
