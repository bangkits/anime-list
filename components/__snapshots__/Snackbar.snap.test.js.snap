// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar Component Snapshot Renders error snackbar component 1`] = `
Object {
  "then": [Function],
}
`;

exports[`Snackbar Component Snapshot Renders success snackbar component 1`] = `
Object {
  "then": [Function],
}
`;

exports[`Snackbar Component Snapshot Renders warning snackbar component 1`] = `
Object {
  "then": [Function],
}
`;
